benchmark/_http-benchmarkers.js

```diff
@@ -0,0 +1,167 @@
+'use strict';
+
+const child_process = require('child_process');
+
+function AutocannonBenchmarker() {
+  this.name = 'autocannon';
+}
+
+AutocannonBenchmarker.prototype.autocannon_exe = process.platform === 'win32'
+                                               ? 'autocannon.cmd'
+                                               : 'autocannon';
+
+AutocannonBenchmarker.prototype.present = function() {
+  const result = child_process.spawnSync(this.autocannon_exe, ['-h']);
+  return !(result.error && result.error.code === 'ENOENT');
+};
+
+AutocannonBenchmarker.prototype.create = function(port, path, duration,
+                                                  connections) {
+  const args = ['-d', duration, '-c', connections, '-j', '-n',
+                `http://127.0.0.1:${port}${path}` ];
+  const child = child_process.spawn(this.autocannon_exe, args);
+  child.stdout.setEncoding('utf8');
+  return child;
+};
+
+AutocannonBenchmarker.prototype.processResults = function(output) {
+  let result;
+  try {
+    result = JSON.parse(output);
+  } catch (err) {
+    // Do nothing, let next line handle this
+  }
+  if (!result || !result.requests || !result.requests.average) {
+    return undefined;
+  } else {
+    return result.requests.average;
+  }
+};
+
+function WrkBenchmarker() {
+  this.name = 'wrk';
+  this.regexp = /Requests\/sec:[ \t]+([0-9\.]+)/;
+}
+
+WrkBenchmarker.prototype.present = function() {
+  const result = child_process.spawnSync('wrk', ['-h']);
+  return !(result.error && result.error.code === 'ENOENT');
+};
+
+WrkBenchmarker.prototype.create = function(port, path, duration, connections) {
+  const args = ['-d', duration, '-c', connections, '-t', 8,
+                `http://127.0.0.1:${port}${path}` ];
+  const child = child_process.spawn('wrk', args);
+  child.stdout.setEncoding('utf8');
+  child.stderr.pipe(process.stderr);
+  return child;
+};
+
+WrkBenchmarker.prototype.processResults = function(output) {
+  const match = output.match(this.regexp);
+  const result = match && +match[1];
+  if (!result) {
+    return undefined;
+  } else {
+    return result;
+  }
+};
+
+const http_benchmarkers = [ new WrkBenchmarker(),
+                            new AutocannonBenchmarker() ];
+
+const supported_http_benchmarkers = [];
+const installed_http_benchmarkers = [];
+var benchmarkers = {};
+
+http_benchmarkers.forEach((benchmarker) => {
+  const name = benchmarker.name;
+  const present = benchmarker.present();
+
+  benchmarkers[name] = {
+    instance: benchmarker,
+    present: present
+  };
+
+  supported_http_benchmarkers.push(name);
+  if (present) {
+    installed_http_benchmarkers.push(name);
+  }
+
+});
+
+let default_http_benchmarker;
+
+if (process.env.NODE_HTTP_BENCHMARKER) {
+  default_http_benchmarker = process.env.NODE_HTTP_BENCHMARKER;
+  if (!benchmarkers[default_http_benchmarker]) {
```

Comment:
I think it would be nice for this to fail here. This way user will get feedback right away (error in configuration), not some time into running all benchmarks.